diff --git a/BUILD.gn b/BUILD.gn
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -339,24 +339,26 @@ if (build_with_chromium) {
     }
   }
-
+
-  # TODO(altimin): this is a temp workaround around very strange link failures
-  # on win-dbg-ng buildbot.
-  if (!is_win) {
-    libtrace_processor_target_type = "source_set"
-  } else {
-    libtrace_processor_target_type = "component"
-  }
-
-  # In Chromium, we want to ensure that we don't link dynamically against sqlite
-  # (as Chromium also uses a more restricted version of sqlite which is actually
-  # shipped to the users).
-  # source_set helps us to achieve that.
-  target(libtrace_processor_target_type, "libtrace_processor") {
-    public_configs = [ "gn:public_config" ]
-    deps = [ "src/trace_processor:lib" ]
-    configs -= [ "//build/config/compiler:chromium_code" ]  # nogncheck
-    configs += [ "//build/config/compiler:no_chromium_code" ]  # nogncheck
-    public_deps = [ "include/perfetto/trace_processor" ]
+  if (enable_perfetto_trace_processor) {
+    # TODO(altimin): this is a temp workaround around very strange link failures
+    # on win-dbg-ng buildbot.
+    if (!is_win) {
+      libtrace_processor_target_type = "source_set"
+    } else {
+      libtrace_processor_target_type = "component"
+    }
+
+    # In Chromium, we want to ensure that we don't link dynamically against sqlite
+    # (as Chromium also uses a more restricted version of sqlite which is actually
+    # shipped to the users).
+    # source_set helps us to achieve that.
+    target(libtrace_processor_target_type, "libtrace_processor") {
+      public_configs = [ "gn:public_config" ]
+      deps = [ "src/trace_processor:lib" ]
+      configs -= [ "//build/config/compiler:chromium_code" ]  # nogncheck
+      configs += [ "//build/config/compiler:no_chromium_code" ]  # nogncheck
+      public_deps = [ "include/perfetto/trace_processor" ]
+    }
   }
   component("perfetto_test_support") {
     testonly = true
